README.md

```diff
@@ -13,207 +13,111 @@ Istanbul's state of the art command line interface, with support for:
-
-2. disable nyc's instrumentation and source-maps, e.g. in `package.json`:
-
-  ```json
-  {
-    "nyc": {
-      "require": [
-        "@babel/register"
-      ],
-      "sourceMap": false,
-      "instrument": false
-    },
-    "scripts": {
-      "test": "cross-env NODE_ENV=test nyc mocha"
-    }
-  }
-  ```
-
-That's all there is to it, better ES2015+ syntax highlighting awaits:
-
-<img width="500" src="screen2.png">
-
-## Support for alternate file extensions (.jsx, .mjs)
-
-Supporting file extensions can be configured through either the configuration arguments or with the `nyc` config section in `package.json`.
-
-```shell
-nyc --extension .jsx --extension .mjs npm test
-```
+#### Adding your overrides
 
 ```json
 {
   "nyc": {
-    "extension": [
-      ".jsx",
-      ".mjs"
-    ]
+    "extends": "@istanbuljs/nyc-config-typescript",
+    "all": true
   }
 }
 ```
+### Configuration files
 
-## Checking coverage
-
-nyc can fail tests if coverage falls below a threshold.
-After running your tests with nyc, simply run:
-
-```shell
-nyc check-coverage --lines 95 --functions 95 --branches 95
-```
-
-nyc also accepts a `--check-coverage` shorthand, which can be used to
-both run tests and check that coverage falls within the threshold provided:
-
-```shell
-nyc --check-coverage --lines 100 npm test
-```
-
-The above check fails if coverage falls below 100%.
-
-To check thresholds on a per-file basis run:
-
-```shell
-nyc check-coverage --lines 95 --per-file
-```
+Any configuration options that can be set via the command line can also be specified in the `nyc` stanza of your package.json, or within a seperate configuration file - a variety of flavors are available:
 
-## Running reports
+| File name       | File Association |
+|-----------------|------------------|
+| `.nycrc`        | JSON             |
+| `.nycrc.json`   | JSON             |
+| `.nycrc.yaml`   | YAML             |
+| `.nycrc.yml`    | YAML             |
+| `nyc.config.js` | CommonJS export  |
 
-Once you've run your tests with nyc, simply run:
+### Common Configuration Options
 
-```bash
-nyc report
-```
+See `npx nyc --help` for all options available.
+You can set these in any of the files listed above, or from the command line.
+This table is a quick TLDR for the rest of this readme and there are more advanced docs available.
 
-To view your coverage report:
+| Option name | Description | Type | Default |
+| ----------- | ----------- | ---- | ------- |
+| `all` | whether or not to instrument all files (not just the ones touched by your test suite) | `Boolean` | `false` |
+| `check-coverage` | check whether coverage is within thresholds, fail if not | `Boolean` | `false` |
+| `extension` | a list of extensions that nyc should attempt to handle in addition to `.js` | `Array<String>` | `['.js']` |
+| `include` | [globs of files to be included from instrumentation](#selecting-files-for-coverage) | `Array<String>` | `['**']`|
```

Comment:
Since this is the same link `exclude` references maybe we should have the same text?  `See [selecting files for coverage] for more info`.